packages/terra-consumer-nav/src/components/nav-items/NavItems.jsx

```diff
@@ -42,14 +46,34 @@ class NavItems extends Component {
     this.handleToggle = this.handleToggle.bind(this);
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (this.props !== nextProps) {
```

Comment:
this is a shallow comparison, which is not good. Should compare attribute directly. Also in order to save indentations should use return statement first.
{code}
if (this.props.currentURL === nextProps.currentURL)
    return;
// The rest of code starts here.
{code}

Reply:
Updated!